src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4848,6 +4853,21 @@ internal void CleanupWindowsPowerShellCompatResources()
             }
         }
 
+        internal static void SyncCurrentLocationHandler(object sender, LocationChangedEventArgs args)
+        {
+            PSSession compatSession = GetWindowsPowerShellCompatRemotingSession();
+            if (compatSession?.Runspace.RunspaceStateInfo.State == RunspaceState.Opened)
```

Comment:
Is it ok to do nothing when either compatSession is null or RunspaceState is not opened?

Reply:
Good observation. That is intentional. If there are problems with the compatSession - they will show up anyway during any WinCompat operation. This handler is called on every location change (which often not related to WinCompat at all) and I don't think it is a good UX if any local directory change would show an error about a broken compatSession.